### DISCUSSION TITLE
Change Download Location

### DISCUSSION THREAD

**Discussion by Spider19996** [bromite/bromite#1814] [Ideas]
Hey, I would like to change the storage location. Unfortunately this does not work. You can only choose between internal memory and SD card. Would it be possible to add a feature that you can also specify a different folder? Or maybe I miss Something?
![Screenshot_20220220-192126](https://user-images.githubusercontent.com/53911689/154858016-23b7bcb3-2f22-4f05-8a7b-32b1af4faaff.jpg)

**Comment by csagan5** [comment]
I think it is not possible due to an Android limitation; @uazo knows better about this.

**Comment by HappyBullfrog** [comment]
The capability you desire already exists. 

The choice you have shown really means "Download to default folder (OR to your own chosen folder) in internal memory's primary partition."

Go to "Settings" – scroll up to "Downloads" listed near the bottom of the page, and then check "Ask me every time".

The next time you download, you can select a folder. 

The folder you specify will be retained and be offered again as the location desired for the next download.

You will also have an opportunity to edit the name of the file you are about to download. 

I use this all the time, and it is very convenient. 

The alternative option shown in your screen capture means "Download onto an external SD card, OR to a second partition in internal memory, if you have created a second partition".

That is, Bromite treats a second partition in internal memory as if it were a SD card.

My OnePlus 8 Pro can accommodate two SIM cards (though I use only one) and allowed me to create an internal second partition. (I used Parallel Space, but this software may have been redundant; I was still learning how to use that device.)

I also have a similar later model, the "OnePlus 8T + 5G from T-Mobile" that can accommodate only one SIM card. So far I have been unable to create an internal second partition by any means.

**Reply by uazo** [reply]
> I think it is not possible due to an Android limitation

I confirm, at least in a10+ before it would be allowed to write where you want.

> The next time you download, you can select a folder.

maybe there is something to check...
I've never seen that choice.

**Comment by Spider19996** [comment]
Thanks for your answers. I tried it again on two different devices. OnePlus 5 with Android 8 (Lineage) and OnePlus 8T with Android 12 (Stock). On both you can not change the location to another folder. (Or I am just too stupid)

When I select "ask each time" I can change the file name but not the folder to save it to. You can only select internal storage or SD and its not possible to specify a new path like /storage/emulated/0/BromiteDownloads for example

Maybe it is an Android restriction. Bromite also only has access to "media files" in the Android permissions while for example the gallery app can read all files on the device.

I have tried a few other browsers. Opera for example lets you select the download folder. You have to select the folder via the Android file manager dialog and allow access and then it is done...
